apps/nascarnextrace/nascarnextrace.star

```diff
@@ -0,0 +1,130 @@
+"""
+Applet: NASCAR Next Race
+Summary: Next NASCAR Race Time and Location - select series
+Description: Shows Time date and location of Next NASCAR Race - Cup, Xfinity, Trucks - based heavily on F1 Next Race from AMillionAir
+Author: jvivona
+"""
+
+load("render.star", "render")
+load("http.star", "http")
+load("encoding/json.star", "json")
+load("time.star", "time")
+load("encoding/base64.star", "base64")
+load("schema.star", "schema")
+load("cache.star", "cache")
+
+DEFAULT_LOCATION = """
+{
+	"lat": "40.6781784",
+	"lng": "-73.9441579",
+	"description": "Brooklyn, NY, USA",
+	"locality": "Brooklyn",
+	"place_id": "ChIJCSF8lBZEwokRhngABHRcdoI",
+	"timezone": "America/New_York"
+}
+"""
+
+#we grab the current schedule from nascar website and cache it at this api location to prevent getting blocked by nascar website, data is refreshed every 30 minutes
+NASCAR_API = "https://tidbyt.apis.ajcomputers.com/nascar/api/"
+DEFAULT_SERIES = "cup"
+
+CONST_VALUES = """
+{
+    "cup" : [ "cup", "#000", "#fff", "NASCAR Cup Series" ],
+    "xfinity" : [ "xfinity", "#4427ad", "#fff", "NASCAR Xfinity Series" ],
+    "trucks" : [ "trucks", "#00669e", "#fff", "Camping World Truck Series" ]
+}
+"""
+
+def main(config):
+    #TIme and date Information
+    location = config.get("location", DEFAULT_LOCATION)
+    loc = json.decode(location)
+    timezone = loc["timezone"]
+    now = time.now().in_location(timezone)
+    Year = now.format("2006")
+    series = config.get("NASCAR_Series", DEFAULT_SERIES)
+
+    NASCAR_DATA = http.get(NASCAR_API + series)
```

Comment:
Even with the ajcomputers API between Tidbyt and the original data source, you might want to add some caching. Keeps request rate down for the API, improves latency and also makes us more likely to render in the face of some networking glitch or say the API restarting, etc.